subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/KotlinPluginAndroidKotlinDSLSmokeTest.groovy

```diff
@@ -45,7 +43,12 @@ class KotlinPluginAndroidKotlinDSLSmokeTest extends AbstractSmokeTest {
 
         when:
         def runner = createRunner(workers, 'clean', ':app:testDebugUnitTestCoverage')
-        def result = useAgpVersion(androidPluginVersion, runner).build()
+        def result = useAgpVersion(androidPluginVersion, runner)
+            .expectLegacyDeprecationWarningIf(kotlinPluginVersion < "1.4.31", deprecationOfConfigurationAsDependency())
```

Comment:
The expectations and versions are already in `KotlinPluginSmokeTest` and `AndroidPluginsSmokeTest`. Can you extract both to fixtures which are Groovy traits and re-use them here?

Reply:
I moved some deprecation logic and messages to `WithDeprecations` trait. I also removed all deprecation logic from `AbstractSmokeTest`. So now deprecation usages are cleaner and not all over the place.

I didn't clean all classes though, just the ones that I touched. Others tests can be cleaned when there will be any change in them.

Reply:
A trait only help if you actually mix in the trait into different types which already have a super-class. I did something like this here: https://github.com/gradle/gradle/pull/19841. The main idea is to add some trait to the runner, so the methods don't need to use the runner as first argument. PTAL!